src/main/java/edu/kis/vh/nursery/FIFORyhmer.java

```diff
@@ -1,21 +1,33 @@
 package edu.kis.vh.nursery;
 
-public class FIFORyhmer extends DefaultCountingOutRyhmer {
+import edu.kis.vh.nursery.stackandlist.IntDataStructure;
 
-    private DefaultCountingOutRyhmer defaultCountingOutRyhmer = new DefaultCountingOutRyhmer();
+public class FIFORyhmer extends DefaultCountingOutRhymer {
+
+    private DefaultCountingOutRhymer defaultCountingOutRhymer = new DefaultCountingOutRhymer();
+
+    public FIFORyhmer(IntDataStructure intDataStructure) {
+        super(intDataStructure);
+    }
+
+    public FIFORyhmer() {
+    }
 
     @Override
     public int countOut() {
         while (!callCheck()) {
-            defaultCountingOutRyhmer.countIn(super.countOut());
+            defaultCountingOutRhymer.countIn(super.countOut());
         }
 
-        int totalAfterCountOut = defaultCountingOutRyhmer.countOut();
+        int totalAfterCountOut = defaultCountingOutRhymer.countOut();
 
-        while (!defaultCountingOutRyhmer.callCheck()) {
-            countIn(defaultCountingOutRyhmer.countOut());
+        while (!defaultCountingOutRhymer.callCheck()) {
+            countIn(defaultCountingOutRhymer.countOut());
         }
 
         return totalAfterCountOut;
     }
 }
+
+//it's better to keep it a a DefaultCountingOutRhymer and make the changes 

```

Comment:
It was not the case for 3.1.14. Maybe if the explaination was tangled with 3.1.18 I would accept.